File: src/Backend/Services/Catalog/Data.SQL/Types/Article/ArticleTypeEntityForList.cs
// Copyright (c) 2023 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

namespace Crib2023.Backend.Services.Catalog.Data.SQL.Types.Article;

/// <summary>
/// Сущность типа "Статья для списка".
/// </summary>
public class ArticleTypeEntityForList
{
    #region Properties

    /// <summary>
    /// Идентификатор.
    /// </summary>
    public long Id { get; set; }

    /// <summary>
    /// Глобальный идентификатор строки.
    /// </summary>
    public Guid RowGuid { get; set; } = Guid.NewGuid();

    /// <summary>
    /// Заголовок.
    /// </summary>
    public string Title { get; set; } = "";

    /// <summary>
    /// Идентификатор экземпляра сущности "Тема".
    /// </summary>
    public long TopicId { get; set; }

    #endregion Properties
}

File: src/Frontend/React/TypeScript/App/src/data/Types/Article/ArticleTypeEntityForList.ts
export interface ArticleTypeEntityForList {
  id: number;
  rowGuid: string;
  title: string;
  topicId: number;
}

export function createArticleTypeEntityForList (options?: Partial<ArticleTypeEntityForList>): ArticleTypeEntityForList {
  return {
    id: options?.id ?? 0,
    rowGuid: options?.rowGuid ?? '',
    title: options?.title ?? '',
    topicId: options?.topicId ?? 0
  };
}
